pkg/hubble/filters/tcp_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package filters
+
+import (
+	"context"
+	"testing"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+func TestFlowTCPFilter(t *testing.T) {
+	var argsevent *v1.Event
+	testflags := []struct {
+		name   string
+		argsf  []*flowpb.FlowFilter
+		argsev *flowpb.TCPFlags
+		want   bool
+	}{
+		{
+			name:   "filterSYNACK___eventSYNACK",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{SYN: true, ACK: true}}}},
```

Comment:
can even go one more layer to remove the need for the `[]*flowpb.FlowFilter{}` wrap

Reply:
Removed the FlowFilter wrap and using only TCPFlags wrap. Looks neat. Thanks. Also have added few more cases to test:

1. Filter={{SYN}, {ACK}}, Event={SYN} .... Verdict: true
2. Filter={{SYN}, {ACK}}, Event={PSH, ACK} .... Verdict: true
3. Filter={{SYN, ACK}}, Event={PSH, ACK} .... Verdict: false
and others...